pkg/kev/config/extensions_test.go

```diff
@@ -0,0 +1,117 @@
+package config_test
+
+import (
+	"bytes"
+
+	"github.com/appvia/kev/pkg/kev/config"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"gopkg.in/yaml.v3"
+)
+
+var _ = Describe("Extentions", func() {
+
+	Describe("parsing", func() {
+		var (
+			parsedCfg  config.K8SServiceConfig
+			k8s        config.K8SServiceConfig
+			extensions = make(map[string]interface{})
+			err        error
+		)
+
+		BeforeEach(func() {
+			k8s = config.K8SServiceConfig{}
+		})
+
+		JustBeforeEach(func() {
+			var buf bytes.Buffer
+			err := yaml.NewEncoder(&buf).Encode(map[string]interface{}{
+				"x-k8s": k8s,
+			})
+			Expect(err).To(Succeed())
+
+			err = yaml.NewDecoder(&buf).Decode(&extensions)
+			Expect(err).To(Succeed())
+		})
+
+		Context("works with defaults", func() {
+			BeforeEach(func() {
+				k8s.Workload.Replicas = 10
+				k8s.Workload.LivenessProbe.Type = config.ProbeTypeNone.String()
+			})
+
+			It("Works with defaults", func() {
+				parsedCfg, err = config.K8SServiceCfgFromMap(extensions)
+				Expect(err).To(Succeed())
+				Expect(parsedCfg).NotTo(BeNil())
+
+				Expect(parsedCfg.Workload.Replicas).To(Equal(10))
+				Expect(parsedCfg.Component).To(BeZero())
+
+				expectedLiveness := config.DefaultLivenessProbe()
+				expectedLiveness.Type = config.ProbeTypeNone.String()
+
+				Expect(parsedCfg.Workload.LivenessProbe).To(BeEquivalentTo(expectedLiveness))
+				Expect(parsedCfg.Workload.ReadinessProbe).To(BeEquivalentTo(config.DefaultReadinessProbe()))
+			})
+		})
+
+		Describe("fails on mandatory fields", func() {
+			Context("missing liveness probe type", func() {
+				BeforeEach(func() {
+					k8s.Workload.Replicas = 10
+				})
+
+				It("returns error", func() {
+					parsedCfg, err = config.K8SServiceCfgFromMap(extensions)
+					Expect(err).To(HaveOccurred())
+					Expect(err.Error()).To(Equal("Workload.LivenessProbe.Type is required"))
+				})
+			})
+
+			Context("missing replicas", func() {
+				BeforeEach(func() {
+					k8s.Workload.LivenessProbe.Type = config.ProbeTypeNone.String()
+				})
+
+				It("returns error", func() {
+					parsedCfg, err = config.K8SServiceCfgFromMap(extensions)
+					Expect(err).To(HaveOccurred())
+					Expect(err.Error()).To(Equal("Workload.Replicas is required"))
+				})
+			})
+		})
+	})
+
+	Describe("Marshalling", func() {
+		It("Invariance", func() {
```

Comment:
minor: would be good to give more description as to what this case is actually testing.